ORM/Repository.php

```diff
@@ -183,6 +183,23 @@ public function execute(Search $search, $resultsType = self::RESULTS_OBJECT)
     }
 
     /**
+     * Delete by query.
+     *
+     * @param Search $search
+     *
+     * @return array
+     */
+    public function deleteByQuery(Search $search)
+    {
+        $results = $this
+            ->manager
+            ->getConnection()
+            ->deleteByQuery($this->types, $search->toArray());
+
+        return $this->parseResult($results, self::RESULTS_RAW, null);
```

Comment:
IMO response could be a bit nicer. Particularly for delete by query we might return:
```
[
"typeName" => [
    "total" => 5,
    "successful" => 5,
    "failed" => 0,
    ],
...
]
```